src/components/Dropdown.js

```diff
@@ -0,0 +1,151 @@
+import { useState, useEffect, useCallback } from "react";
+import SVGIcon from "@/src/components/SVGIcon";
+import {
+  DropdownContainer,
+  DropdownLabel,
+  DropdownButton,
+  DropdownMenu,
+  DropdownItem,
+  ErrorMessage,
+  IconWrapper,
+} from "@/src/components/styles/Dropdown";
+
+export default function Dropdown({
+  label,
+  name,
+  options,
+  buttonText,
+  onOptionClick,
+  required,
+  errorMessage,
+  defaultSelected,
+}) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedOption, setSelectedOption] = useState(defaultSelected || "");
+  const [focusedIndex, setFocusedIndex] = useState(-1);
+
+  useEffect(() => {
+    function handleClickOutside(event) {
+      if (isOpen && !event.target.closest(".dropdown-container")) {
+        setIsOpen(false);
+      }
+    }
+
+    document.addEventListener("mousedown", handleClickOutside);
```

Comment:
the "mousedown" event is only relevant for desktop devices, for mobile devices you need to handle touch events.